93 result visualization 1

along4 opened the pull request:
Developing tools to visualize and plot SAMMY results. 

There are two main files of interest that SAMMY prints out, the SAMMY.LPT file and the SAMMY.LST file. 
The LPT file contains all the information around the fitting parameters of the sammy run
The LST file contain all the data used for the fit (cross-sections or transmission), along with the final fitted result.  

To Do: 
- [x] Build out class to hold result information. Use already built classes like nuclearParameters and experimentalParameters.
- [x] Build manager that aggregates all the results from a given sammy fit.

along4 commented:
Hey Chen, I should be all done with the edits for results processing and visualization. Let me know if you have any questions.

along4 commented:
Hey @KedoKudo, I realized that I did not add any unit tests for the new modules that I wrote. I will try to do this tonight.

KedoKudo commented:
> Hey @KedoKudo, I realized that I did not add any unit tests for the new modules that I wrote. I will try to do this tonight.

That would be great, thanks. I just finished re-processing all the new resonance data collected at VENUS with TPX1. I will start reviewing the PR tomorrow.

KedoKudo reviewed (changes requested):
Please address the suggestion provided by copilot and consider the suggested alternative methods for access member availability and plotting function APIS.
Also, some of the the function does not have docstring while others do, a simple pass with copilot should be able to address them quickly.

along4 commented:
Hey @KedoKudo, I have implemented the updates based on the suggestions given. Let me know if you have any questions during further review. Below is a overview of the major and minor changes made during this pull request. 


This pull request introduces several updates across the codebase, focusing on improving flexibility in data models, enhancing functionality for SAMMY data handling, and fixing minor issues. The most significant changes include making fields in multiple data models optional, adding a new `SammyData` class for handling SAMMY `.LST` files, and fixing a typo in field names. Below is a categorized summary of the most important changes:

### SAMMY Data Handling Enhancements
* Renamed `dataParameters` to `SammyData`, enabling structured handling of SAMMY `.LST` files. This includes methods for loading and validating SAMMY results from `SAMMY.LPT` and `SAMMY.LST` files.
* Updated `FitConfig` to use the name `SammyData` class instead of `dataParameters`. [[1]](diffhunk://#diff-322cfbd28d150d8a28650d7e6572478b244370f325c0d75501be20f247e7a88fL34-R34) [[2]](diffhunk://#diff-322cfbd28d150d8a28650d7e6572478b244370f325c0d75501be20f247e7a88fL55-R53)

#### New Features
* Added a new `ResultsMap` model in `results/models.py` to encapsulate data related to pixel information, isotope abundances, and background/normalization data.
* Added new 'plotter.py` module to handle all plotting functionality with sammy results. Functions in this module are passed `RunResults` objects to be processed and plotted. 

### Notebook Updates
* Added a warning to the `pleiades_sammy_parFile.ipynb` notebook indicating it has not been updated for v2.0 changes.
* Updated the `pleiades_Si_transmission.ipynb` to reflect the usage of the plotter module created in `/pleiades/sammy/results/`

#### Minor Bug Fixes
* Corrected a typo in field names from `isotope_infomation` to `isotope_information` in `IsotopeParameters` and related references. [[1]](diffhunk://#diff-72f57b9ad4180e0579a823c0a696fa2bceabf8d95ecf5feeae046f711794ad52L116-R116) [[2]](diffhunk://#diff-c679b9a7490e436d83fd27a6c30635e0a070bb201bbf4e65a05443c32860651aL403-R428)
* Corrected typo with name of `sammyData` to `SammyData`

#### Minor Data Model Updates
* Updated `IsotopeMassData`, `IsotopeInfo`, `RadiusParameters`, and `IsotopeParameters` models to make fields optional by adding default values of `None`. This increases flexibility when working with partial data. [[1]](diffhunk://#diff-b9f8b345507106369c3707d3e5c3b015da6de7b3cd50434f1df539e1897583efL51-R54) [[2]](diffhunk://#diff-b9f8b345507106369c3707d3e5c3b015da6de7b3cd50434f1df539e1897583efL84-R89) [[3]](diffhunk://#diff-c679b9a7490e436d83fd27a6c30635e0a070bb201bbf4e65a05443c32860651aL109-R120) [[4]](diffhunk://#diff-c679b9a7490e436d83fd27a6c30635e0a070bb201bbf4e65a05443c32860651aL290-R293)

KedoKudo commented on src/pleiades/utils/plotter.py:
Do we still need this file?

KedoKudo reviewed (approved):
@along4 Can you confirm if we still need the placeholder file (see comments)? If yes, I will merge the PR. Otherwise, please remove it. Thanks 🙏
